google/cloud/spanner/samples/postgresql_samples.cc

```diff
@@ -557,28 +559,106 @@ void InformationSchema(
+  std::cout << "Inserted " << dml_result.RowsModified() << " venue(s)"
+            << " with NULL revenue.\n";
+
+  // Insert a Venue with a NaN value for the Revenue column.
+  commit = client.Commit(
+      [&client, &dml_result](google::cloud::spanner::Transaction txn)
+          -> google::cloud::StatusOr<google::cloud::spanner::Mutations> {
+        auto sql = google::cloud::spanner::SqlStatement(
+            R"""(
+                INSERT INTO Venues (VenueId, Name, Revenue)
+                    VALUES ($1, $2, $3)
+            )""",
+            {{"p1", google::cloud::spanner::Value(3)},
+             {"p2", google::cloud::spanner::Value("Venue 3")},
+             {"p3",
+              google::cloud::spanner::Value(
+                  google::cloud::spanner::MakePgNumeric("NaN").value())}});
+        auto insert = client.ExecuteDml(std::move(txn), std::move(sql));
+        if (!insert) return std::move(insert).status();
+        dml_result = *std::move(insert);
+        return google::cloud::spanner::Mutations{};
+      });
+  if (!commit) throw std::runtime_error(commit.status().message());
+  std::cout << "Inserted " << dml_result.RowsModified() << " venue(s)"
+            << " with NaN revenue.\n";
+
+  // Mutations can also be used to insert/update values, including NaNs.
+  auto insert_venues =  //
+      google::cloud::spanner::InsertMutationBuilder(
+          "Venues", {"VenueId", "Name", "Revenue"})
+          .EmplaceRow(4, "Venue 4",
+                      google::cloud::spanner::MakePgNumeric("125.10").value())
+          .EmplaceRow(5, "Venue 5",
+                      google::cloud::spanner::MakePgNumeric("NaN").value())
+          .Build();
+  commit = client.Commit(google::cloud::spanner::Mutations{insert_venues});
+  if (!commit) throw std::runtime_error(commit.status().message());
+  std::cout << "Inserted 2 venues using mutations at "
+            << commit->commit_timestamp << ".\n";
+
+  // Get all Venues and inspect the Revenue values.
   auto sql = google::cloud::spanner::SqlStatement(R"""(
-      SELECT Name, Revenues FROM Venues
+      SELECT Name, Revenue FROM Venues
   )""");
-  using RowType =
-      std::tuple<std::string, absl::optional<google::cloud::spanner::Numeric>>;
+  using RowType = std::tuple<std::string,
+                             absl::optional<google::cloud::spanner::PgNumeric>>;
   auto rows = client.ExecuteQuery(std::move(sql));
   for (auto const& row : google::cloud::spanner::StreamOf<RowType>(rows)) {
     if (!row) throw std::runtime_error(row.status().message());
-    if (!std::get<1>(*row).has_value()) continue;
-    std::cout << "Revenues of " << std::get<0>(*row) << ": "
-              << *std::get<1>(*row) << "\n";
+    std::cout << "Revenue of " << std::get<0>(*row) << ": ";
+    if (std::get<1>(*row).has_value()) {
+      std::cout << *std::get<1>(*row);
+    } else {
+      std::cout << "NULL";
+    }
+    std::cout << "\n";
```

Comment:
Seems to me that 100 lines is a tad long for an example, but I assume that is the requirement. 🤷‍♂️

Reply:
Yes, I agree.  But yes, I don't make the rules.